src/mgr/MgrStandby.cc

```diff
@@ -41,7 +41,8 @@ MgrStandby::MgrStandby(int argc, const char **argv) :
   monc{g_ceph_context, poolctx},
   client_messenger(Messenger::create(
 		     g_ceph_context,
-		     cct->_conf.get_val<std::string>("ms_type"),
+		     cct->_conf.get_val<std::string>("ms_public_type").empty() ?
```

Comment:
could you please expand a little bit on why we need to prefer to "ms_public_type" and fall back to "ms_type" if the former is empty?

i think the reason is likely to be consistent with the description of the fact that the messenger is used to talk the public facing addresses.

Reply:
i was testing rdma stack with `ms_public_type` set to `async+rdma` explicitly and left `ms_type` & `ms_cluster_type` to their defaults. all ceph daemons work as expected expect mgr.

i think this is a bugfix more than a cleanup.